### ISSUE TITLE
Tests: Run tests that can be run in parallel in parallel

### ISSUE THREAD

**Issue by karuppiah7890** [kubernetes-sigs/cluster-api-provider-azure#1775]
There are many Golang tests in the code base that are not run in parallel. For example, tests run using `t.Run()`. Also, even tests containing multiple `t.Run()` can be run in parallel at the top level, using `t.Parallel()`. Most times I think it would just work - unless there's some shared data across tests, which there shouldn't be ideally, so that tests can run in parallel. So as part of this issue, we can ensure that every test that can be in parallel (top level or t.Run()) is run in parallel

**Comment by VibhorChinda** [comment]
I want to help here :)) 

We just need to change t.run --> t.Parallel ??
Or there is something more to this @CecileRobertMichon

**Comment by CecileRobertMichon** [comment]
@VibhorChinda yes this would mean auditing the existing unit tests and changing the ones that don't run in parallel to parallel! We should also make sure there are no race conditions in how the test cases are set up, #2646 could come in handy for this. 

A good intro to this: https://eleni.blog/2019/05/11/parallel-test-execution-in-go/ (pay close attention to the "tip" at the end specifically, we need to make sure we are capturing the range variable for all our parallel tests).

Please comment `/assign` if you will be working on this.

**Comment by CecileRobertMichon** [comment]
/remove-lifecycle rotten
/unassign @VibhorChinda 

@VibhorChinda please re-assign yourself if you intend to keep working on this.

**Comment by CecileRobertMichon** [comment]
@larsknutson sure, please comment "/assign" to have the bot assign you to the issue

**Comment by nawazkh** [comment]
This has been deep hidden! 
How is it going for you @larsknutson ?
